demo/richards/richards_driver.c

```diff
@@ -37,8 +35,10 @@ int main(int argc, char **argv) {
   ierr = TDySetFromOptions(tdy); CHKERRQ(ierr);
 
   ierr = TDyDriverInitializeTDy(tdy); CHKERRQ(ierr);
-  if (!rank) tdy->io->io_process = PETSC_TRUE;
-  tdy->io->print_intermediate = print_intermediate;
+  if (!rank) {
```

Comment:
Not a big deal, but if you are using `!rank` here to mean `rank 0`, then `if (rank == 0)` is more clear.

Reply:
I don't disagree, but this is a PETSc convention that Barry felt very strongly about so we (PETSc developers) always write `!rank`.

Reply:
I simply followed the existing code and don't have a strong opinion either way. One more reason to have a style guide.